lib/discourse_theme/scaffold.rb

```diff
@@ -1,75 +1,119 @@
-      desktop/head_tag.html
-      desktop/body_tag.html
-
-      mobile/mobile.scss
-      mobile/header.html
-      mobile/after_header.html
-      mobile/footer.html
-      mobile/head_tag.html
-      mobile/body_tag.html
 
       locales/en.yml
 
       settings.yml
     }
 
-    ABOUT_JSON = <<~STR
-    {
-      "name": "#NAME#",
-      "about_url": null,
-      "license_url": null,
-      "assets": {
-      },
-      "color_schemes": {
-      }
+    ABOUT_JSON = {
+      about_url: nil,
+      license_url: nil,
+      assets: {}
     }
-  STR
 
     HELP = <<~STR
       Are you a bit lost? Be sure to read https://meta.discourse.org/t/how-to-develop-custom-themes/60848
     STR
 
     GIT_IGNORE = <<~STR
-    .discourse-site
-    HELP
-  STR
+      .discourse-site
+      node_modules
+      HELP
+    STR
+
+    API_INITIALIZER = <<~STR
+      import { apiInitializer } from "discourse/lib/api";
+
+      export default apiInitializer("0.11.1", api => {
+        console.log("hello world from api initializer!");
+      });
+    STR
+
+    PACKAGE_JSON = <<~STR
+      {
+        "author": "#AUTHOR",
+        "license": "MIT",
+        "devDependencies": {
+          "eslint-config-discourse": "latest"
+        }
+      }
+    STR
+
+    ESLINT_RC = <<~STR
+      {
+        "extends": "eslint-config-discourse"
+      }
+    STR
+
+    TEMPLATE_LINT_RC = <<~STR
+      module.exports = {
+        plugins: ["ember-template-lint-plugin-discourse"],
+        extends: "discourse:recommended",
+      };
+    STR
+
+    EN_YML = <<~STR
+      en:
+    STR
 
     def self.generate(dir)
-      Cli.progress "Generating a scaffold theme at #{dir}"
+      UI.progress "Generating a scaffold theme at #{dir}"
 
-      name = Cli.ask("What would you like to call your theme?").strip
+      name = UI.ask("What would you like to call your theme?").strip
+      is_component = UI.yes?("Is this a component?")
 
       FileUtils.mkdir_p dir
       Dir.chdir dir do
-        File.write('about.json', ABOUT_JSON.sub("#NAME#", name))
+        UI.info "Creating about.json"
+        about_template = ABOUT_JSON.dup
+        about_template[:name] = name
+        if is_component
+          about_template[:component] = true
+        else
+          about_template[:color_schemes] = {}
+        end
+        File.write('about.json', JSON.pretty_generate(about_template))
+
+        UI.info "Creating HELP"
         File.write('HELP', HELP)
+
+        UI.info "Creating package.json"
+        author = UI.ask("Who is authoring the theme?").strip
+        File.write('package.json', PACKAGE_JSON.sub("#AUTHOR", author))
+
+        UI.info "Creating .template-lintrc.js"
+        File.write('.template-lintrc.js', TEMPLATE_LINT_RC)
+
+        UI.info "Creating .eslintrc"
+        File.write('.eslintrc', ESLINT_RC)
+
+        UI.info "Creating .gitignore"
         File.write('.gitignore', GIT_IGNORE)
 
+        initializer = "javascripts/discourse/api-initializers/#{name}.js"
```

Comment:
Should this be under the `/discourse` namespace? Or should it be under `/my-theme-name`? (either work... so not sure which we should be encouraging)

Reply:
I personally use discourse, and like it better as it's following core pattern, but I don't mind